Client/Client.cs

```diff
@@ -127,78 +127,126 @@ public Task<DeltaQueryResult> DeltaQuery(string stateToken)
             {
                 foreach (string parameter in propertyList)
                 {
-                   options.Add( new QueryOption("$select", parameter));
+                    options.Add(new QueryOption("$select", parameter));
                 }
             }
-            
-            //run graph query
-            var graphResult = await graphServiceClient.Me.MailFolders.Delta().Request(options).GetAsync();
-            List<Dictionary<string, object>> resultObjectList = new List<Dictionary<string, object>>();
 
-            if (stateToken == null)
+            //run graph query
+            IMailFolderDeltaCollectionPage graphResult;
+            try
+            {
+                graphResult = await graphServiceClient.Me.MailFolders.Delta().Request(options).GetAsync();
```

Comment:
This can throw exceptions, in particular Msal exceptions. Some of these need to be processed.